src/Microsoft.ML.Onnx/OnnxUtils.cs

```diff
@@ -308,14 +308,38 @@ public ModelArgs(string name, TensorProto.Types.DataType dataType, List<long> di
                 case DataKind.TX:
                     dataType = TensorProto.Types.DataType.String;
                     break;
+                case DataKind.I1:
+                    dataType = TensorProto.Types.DataType.Int8;
+                    break;
+                case DataKind.U1:
+                    dataType = TensorProto.Types.DataType.Uint8;
+                    break;
+                case DataKind.I2:
+                    dataType = TensorProto.Types.DataType.Int16;
+                    break;
+                case DataKind.U2:
+                    dataType = TensorProto.Types.DataType.Uint16;
+                    break;
+                case DataKind.I4:
+                    dataType = TensorProto.Types.DataType.Int32;
+                    break;
                 case DataKind.U4:
                     dataType = TensorProto.Types.DataType.Int64;
```

Comment:
Should this be a `Uint32`? Or is there no support for a Uint32, and we're storing in a Int64?

@TomFinley

Reply:
This is not a part of this PR. That mapping is used across the last two Windows ML releases.

Reply:
I opened an issue to discuss/track: https://github.com/dotnet/machinelearning/issues/1198